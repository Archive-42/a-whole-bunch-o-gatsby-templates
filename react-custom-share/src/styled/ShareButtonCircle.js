import { cx, css } from 'emotion';
import PropTypes from 'prop-types';
import React from 'react';

import colors from './colors';
import ShareButton from '../ShareButton';

let styles = css`
  align-items: center;
  border: 0;
  border-radius: 50%;
  cursor: pointer;
  display: flex;
  height: 44px;
  justify-content: center;
  margin: 3px;
  position: relative;
  transition: 0.3s;
  width: 44px;

  &::before {
    background-color: rgba(255, 255, 255, 0.2);
    border-bottom-left-radius: 100px;
    border-top-left-radius: 100px;
    color: inherit;
    content: '';
    height: 100%;
    left: 50%;
    position: absolute;
    top: 50%;
    transform: translate(-100%, -50%);
    transition: 0.2s;
    width: 50%;
  }

  & > svg {
    fill: #fff;
    height: auto;
    transform: scale(1);
    transition: 0.3s;
    width: 22px;
    z-index: 1;
  }

  &:hover {
    &::before {
      background-color: rgba(0, 0, 0, 0.1);
      transform: translate(0%, -50%) rotateY(180deg);
    }

    & > svg {
      transform: scale(1.2);
    }
  }

  @media (any-hover: none) {
    &:hover {
      &::before {
        background-color: rgba(255, 255, 255, 0.2);
        transform: translate(-100%, -50%);
      }

      & > svg {
        transform: scale(1);
      }
    }
  }

  @media (min-width: 330px) {
    margin: 5px;
  }

  @media (min-width: 600px) {
    margin: 7px;
  }

  @media (min-width: 1025px) {
    margin: 10px;
  }
`;

const dynamicStyles = props => css`
  background: ${colors[props.network]};
`;

const ShareButtonCircle = props => {
  const { className, ...rest } = props;
  const dynamicStylesStr = dynamicStyles(props);

  return (
    <ShareButton
      className={cx(styles, dynamicStylesStr, className)}
      {...rest}
    />
  );
};

ShareButtonCircle.propTypes = {
  className: PropTypes.any,
};

export default ShareButtonCircle;
