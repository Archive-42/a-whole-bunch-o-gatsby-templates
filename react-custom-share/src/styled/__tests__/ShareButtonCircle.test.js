import { createSerializer } from 'jest-emotion';
import { render, cleanup } from 'react-testing-library';
import * as emotion from 'emotion';
import FaTwitter from 'react-icons/lib/fa/twitter';
import React from 'react';

import ShareButtonCircle from '../ShareButtonCircle';

expect.addSnapshotSerializer(createSerializer(emotion));

afterEach(cleanup);

describe('<ShareButtonCircle>', () => {
  test(`renders unchanged`, () => {
    const props = { url: 'https://website-to-share.com', network: 'Twitter' };

    const { container } = render(
      <ShareButtonCircle {...props}>
        <FaTwitter />
      </ShareButtonCircle>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
