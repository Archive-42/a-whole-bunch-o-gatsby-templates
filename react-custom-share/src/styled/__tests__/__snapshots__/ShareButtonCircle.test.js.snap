// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShareButtonCircle> renders unchanged 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 0;
  border-radius: 50%;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 44px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 3px;
  position: relative;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  width: 44px;
  background: rgba(99,205,241,1);
}

.emotion-0::before {
  background-color: rgba(255,255,255,0.2);
  border-bottom-left-radius: 100px;
  border-top-left-radius: 100px;
  color: inherit;
  content: '';
  height: 100%;
  left: 50%;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-100%,-50%);
  -ms-transform: translate(-100%,-50%);
  transform: translate(-100%,-50%);
  -webkit-transition: 0.2s;
  transition: 0.2s;
  width: 50%;
}

.emotion-0 > svg {
  fill: #fff;
  height: auto;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-transition: 0.3s;
  transition: 0.3s;
  width: 22px;
  z-index: 1;
}

.emotion-0:hover::before {
  background-color: rgba(0,0,0,0.1);
  -webkit-transform: translate(0%,-50%) rotateY(180deg);
  -ms-transform: translate(0%,-50%) rotateY(180deg);
  transform: translate(0%,-50%) rotateY(180deg);
}

.emotion-0:hover > svg {
  -webkit-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

@media (any-hover:none) {
  .emotion-0:hover::before {
    background-color: rgba(255,255,255,0.2);
    -webkit-transform: translate(-100%,-50%);
    -ms-transform: translate(-100%,-50%);
    transform: translate(-100%,-50%);
  }

  .emotion-0:hover > svg {
    -webkit-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }
}

@media (min-width:330px) {
  .emotion-0 {
    margin: 5px;
  }
}

@media (min-width:600px) {
  .emotion-0 {
    margin: 7px;
  }
}

@media (min-width:1025px) {
  .emotion-0 {
    margin: 10px;
  }
}

<button
  aria-label="Share on Twitter"
  class="emotion-0 rcs-button"
>
  <svg
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style="vertical-align: middle;"
    viewBox="0 0 40 40"
    width="1em"
  >
    <g>
      <path
        d="m37.7 9.1q-1.5 2.2-3.7 3.7 0.1 0.3 0.1 1 0 2.9-0.9 5.8t-2.6 5.5-4.1 4.7-5.7 3.3-7.2 1.2q-6.1 0-11.1-3.3 0.8 0.1 1.7 0.1 5 0 9-3-2.4-0.1-4.2-1.5t-2.6-3.5q0.8 0.1 1.4 0.1 1 0 1.9-0.3-2.5-0.5-4.1-2.5t-1.7-4.6v0q1.5 0.8 3.3 0.9-1.5-1-2.4-2.6t-0.8-3.4q0-2 0.9-3.7 2.7 3.4 6.6 5.4t8.3 2.2q-0.2-0.9-0.2-1.7 0-3 2.1-5.1t5.1-2.1q3.2 0 5.3 2.3 2.4-0.5 4.6-1.7-0.8 2.5-3.2 3.9 2.1-0.2 4.2-1.1z"
      />
    </g>
  </svg>
</button>
`;
